File: app/src/main/java/wangdaye/com/geometricweather/daily/adapter/holder/AirQualityHolder.java
package wangdaye.com.geometricweather.daily.adapter.holder;

import android.annotation.SuppressLint;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.core.graphics.ColorUtils;

import wangdaye.com.geometricweather.R;
import wangdaye.com.geometricweather.common.basic.models.weather.AirQuality;
import wangdaye.com.geometricweather.common.ui.widgets.RoundProgress;
import wangdaye.com.geometricweather.daily.adapter.DailyWeatherAdapter;
import wangdaye.com.geometricweather.daily.adapter.model.DailyAirQuality;

public class AirQualityHolder extends DailyWeatherAdapter.ViewHolder {

    private final RoundProgress mProgress;
    private final TextView mContent;

    public AirQualityHolder(ViewGroup parent) {
        super(LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_weather_daily_air, parent, false));
        mProgress = itemView.findViewById(R.id.item_weather_daily_air_progress);
        mContent = itemView.findViewById(R.id.item_weather_daily_air_content);
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onBindView(DailyWeatherAdapter.ViewModel model, int position) {
        AirQuality airQuality = ((DailyAirQuality) model).getAirQuality();

        int aqi = airQuality.getAqiIndex();
        int color = airQuality.getAqiColor(itemView.getContext());

        mProgress.setMax(400);
        mProgress.setProgress(aqi);
        mProgress.setProgressColor(color);
        mProgress.setProgressBackgroundColor(
                ColorUtils.setAlphaComponent(color, (int) (255 * 0.1))
        );

        mContent.setText(aqi + " / " + airQuality.getAqiText());
    }
}

File: app/src/main/java/org/breezyweather/daily/adapter/holder/AirQualityHolder.kt
package org.breezyweather.daily.adapter.holder

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.TextView
import androidx.core.graphics.ColorUtils
import org.breezyweather.R
import org.breezyweather.common.ui.widgets.RoundProgress
import org.breezyweather.daily.adapter.DailyWeatherAdapter
import org.breezyweather.daily.adapter.model.DailyAirQuality

class AirQualityHolder(parent: ViewGroup) : DailyWeatherAdapter.ViewHolder(
    LayoutInflater.from(parent.context).inflate(R.layout.item_weather_daily_air, parent, false)
) {
    private val mProgress: RoundProgress = itemView.findViewById(R.id.item_weather_daily_air_progress)
    private val mContent: TextView = itemView.findViewById(R.id.item_weather_daily_air_content)

    @SuppressLint("SetTextI18n")
    override fun onBindView(model: DailyWeatherAdapter.ViewModel, position: Int) {
        val airQuality = (model as DailyAirQuality).airQuality
        val aqi = airQuality.getIndex()
        val color = airQuality.getColor(itemView.context)
        mProgress.apply {
            max = 400f
            if (aqi != null) {
                progress = aqi.toFloat()
            }
            setProgressColor(color)
            setProgressBackgroundColor(ColorUtils.setAlphaComponent(color, (255 * 0.1).toInt()))
        }
        mContent.text = aqi.toString() + " / " + airQuality.getName(itemView.context)
    }
}